src/base/castlerandom.pas

```diff
@@ -47,11 +47,12 @@   TCastleRandom = class(TObject)
     function RandomBoolean: boolean;
     { Randomly provides "-1", "0" or "1" with equal chances. }
     function RandomSign: longint;
-    { Returns a random number in 0 .. High(LongWord) range. }
+    { Returns a random number in 1 .. High(LongWord) range. }
```

Comment:
Another technical (and might be much more serious) problem of single-variable xorshift is that it has only ONE random sequence 2^32-1 integers long. Random seed just takes the place to start this random sequence. So, e.g. we can get seed[i] and seed[i+1] for 2 our random generators and the sequences they generate will be absolutely equal except for the first element in the first sequence, while seed[i] and seed[i+1] have nothing in common.
A simple example of the problem might be generating two 100x100 simple maps which have ~1000/2^32 chance to contain obviously identical pattern - much higher than it should be normally, but still too low to care about :).
An additional by-side of this fact is that 2 randoms in a sequence cannot be equal unless a full 2^32-1 cycle has passed. This problem is also solved in variable-cycling algorithms. E.g. `TCastleRandom.Random32bit = TCastleRandom.Random32bit` will never be true no matter how many times it is asked (unless accidently seed=0).
If we ask for integer randoms we use `mod` and therefore unless integer is max(DWord) we can get both zero values and identical values.